pyproject.toml

```diff
@@ -52,8 +54,25 @@ norecursedirs = [
     "share",
     ".tox",
     ".git",
-    "docs"
+    "docs",
 ]
 
 [tool.mypy]
-exclude = ["venv"]
\ No newline at end of file
+exclude = [".venv", "build", "noxfile.py", "tests"]
+
+[tool.ruff]
+line-length = 127 # The GitHub editor is 127 chars wide
+
+
+[dependency-groups]
+dev = [
+    "nox",
+    "ruff>=0.11.7",
+    "hypothesis",
+    "pytest",
+    "pytest-cov",
+    "mypy",
+    "sphinx",
+]
+
+jupyter = ["ipykernel", "pip"]
```

Comment:
just curious - kinda unrelated - but have you tried running the Jupyter notebook? if so, is it all still working?